### ISSUE TITLE
[BUG] False positive in `no-contradicting-classname`  rule when using custom gradient values

### ISSUE THREAD

**Issue by abdul-alhasany** [francoismassart/eslint-plugin-tailwindcss#271]
**Describe the bug**
When using the code below I get an error: `Classnames to-[2.5px], to-transparent are conflicting!` and also `Classnames via-[2px], via-transparent are conflicting!`.

Both messages are incorrect since[ tailwind allows for arbitrary values](https://tailwindcss.com/docs/gradient-color-stops#arbitrary-values) using the same syntax for color and position: `from|via|to-{color|position}`

**To Reproduce**
```html
<div class="bg-gradient
	from-primary
	via-[2px]
	via-transparent
	to-[2.5px]
	to-transparent"
/>
```
**Expected behavior**
No error messages

**Environment (please complete the following information):**
 - OS: Windows 11
 - Softwares + version used:
     -   VSCode 1.81.1
     -  npm 9.7.1
     - node 18.15

**Additional context**
N/A

**eslint config file or live demo**
```js
module.exports = {
	root: true,
	extends: [
		'@kalimahapps',
		'plugin:tailwindcss/recommended',
	],
	plugins: ['tailwindcss', '@kalimahapps/eslint-plugin-tailwind'],
	rules: {
		'tailwindcss/classnames-order': 'off',
		'@kalimahapps/tailwind/sort': 'warn',
		'@kalimahapps/tailwind/multiline': 'warn',
	},
};
```

**Comment by elussich-globant** [comment]
Hi @abdul-alhasany!

Thanks for creating this bug. I'm experiencing the same issue in one of our projects.

Two things I want to comment on:
- the issue will come and go, depending on the `to-<color>` value; for example, if I set it to `to-red`, the error will disappear, but if I set it to `to-white` the error persists 🤔 
- as a possible workaround, I set a custom value in my `tailwind.config.js` file, in the `colors` definitions, named `'custom-transparent'` (could be any name, fwiw), with a value of `'rgba(0, 0, 0, 0)'`, and then, when set it inline as `to-custom-transparent` the error goes away 🤷 

Also, for the record, I've tried upgrading the package to a beta version, as recommended here in [this ticket](https://github.com/francoismassart/eslint-plugin-tailwindcss/issues/260#issuecomment-1743089972), but the issue does not get resolved (I wonder if it has to do with gradient values specifically).

My two cents!

Hopefully this can get resolved soon.
Thanks!

**Comment by francoismassart** [comment]
Fixed in v4, it will be release after cleaning up the existing issues.
